public/js/shell.js

```diff
@@ -24,8 +24,8 @@ export default class Shell {
     this._states = new Map();
   }
 
-  addState(page, state) {
-    this._states.set(page, state);
+  addState(route, state) {
```

Comment:
suggestions: 

* rename `addState` to `setStateForRoute`
* rename `state` to `shellState`